source/linux/Foundational_Components/U-Boot/UG-QSPI.rst

```diff
@@ -279,83 +516,124 @@ documented in U-boot SPI section.
       => tftp ${loadaddr} u-boot.img
       => sf update $loadaddr 0x300000 $filesize
 
-    **phy calibration**
+    4. Change boot mode pins to OSPI boot mode and reboot AM64x
 
-    phy calibration allows for higher read performance. to enable phy, the phy
-    calibration pattern must be flashed to ospi at the start of the last erase
-    sector. for the cypress s28hs512t flash, this lies at the address 0x3fc0000.
+    **Flashing images to OSPI NOR using SD card**
 
-    download the binary file containing the phy pattern from :download:`here </files/ospi_phy_pattern>`.
-    below commands can be used to flash the phy pattern, with the location of the
-    pattern depending on which flash is being used:
+    In this example, load binaries from SD card. Assume bootloader names are
+    'tiboot3.bin', 'tispl.bin', 'u-boot.img'. Boot via SD card boot and stop at
+    U-boot prompt before procceeding.
+
+    1. Use fatload command to load the bootloaders into memory and flash to
+    OSPI NOR
+
+    .. code-block:: console
+
+      => sf probe
+      => fatload mmc 1 ${loadaddr} tiboot3.bin
+      => sf update $loadaddr 0x0 $filesize
+      => fatload mmc 1 ${loadaddr} tispl.bin
+      => sf update $loadaddr 0x100000 $filesize
+      => fatload mmc 1 ${loadaddr} u-boot.img
+      => sf update $loadaddr 0x300000 $filesize
+
+    2. Change boot mode pins to OSPI boot mode and reboot AM64x
+
+    **OSPI Boot Mode**
+
+    Please refer to the AM64x TRM Section 4.4 for more information. The
+    following command could also be used to change boot mode to OSPI after
+    following one of the methods above. Verify to use the "reset" command which
+    uses warm reset and not cold reset.
+
+    .. code-block:: console
+
+      => mw.l 0x43000030 0x00000273; reset
+
+    **Phy Calibration**
+
+    Phy Calibration allows for higher read performance. To enable phy, the phy
+    calibration pattern must be flashed to OSPI at the start of the last erase
+    sector. For the Cypress S28HS512T flash, this lies at the address 0x3FC0000.
+
+    Download the binary file containing the phy pattern from :download:`here </files/ospi_phy_pattern>`.
+    The commands below can be used to flash the phy pattern, with the location
+    of the pattern depending on which flash is being used:
 
     .. code-block:: console
 
        => sf probe
        => tftp ${loadaddr} ospi_phy_pattern
        => sf update $loadaddr 0x3fc0000 $filesize
 
-    **flash layout for ospi**
+    **Flash layout for OSPI NOR**
 
     .. code-block:: console
```

Comment:
Honestly surprised sphinx can parse this code block as is. It goes against the spec definition and my syntax highlighting fails to parse it. It's supposed to be that the first line of content in the block actually dictates the indention level used for the rest of the block, but this starts with the line with the most whitespace.

You could move the addresses to the right side of the graph to make this more resilient to parser changes. But it's fine for now.

```suggestion
    .. code-block:: text
```